platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/NetworkUtilsTest.java

```diff
@@ -40,4 +59,42 @@ void handleNetworkExceptionTest() {
                 () -> NetworkUtils.handleNetworkException(new InterruptedException(), null),
                 "an interrupted exception should be rethrown");
     }
+
+    /**
+     * Tests that given a list of valid Swirlds production certificates,
+     * {@link NetworkUtils#identifyTlsPeer(Certificate[], List)} is able to successfully identify a matching peer.
+     */
+    @Test
+    void testExtractPeerInfoWorksForMainnet()
+            throws URISyntaxException, KeyLoadingException, KeyStoreException, InvalidAlgorithmParameterException {
+
+        // sample node names from mainnet
+        final List<String> names =
```

Comment:
why just 10 nodes? why not all of them?

Reply:
There are 157 of those certificate entries in main-net. For this test, having a few peers get identified is sufficient to prove the logic works (since all the certificates are generated exactly the same way)

Reply:
I've updated the tests to use all of them